### PR TITLE
Add hide option for health status bar

### PR DESCRIPTION
**Pull request by reginaelyse**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

A few Red Hat folks requested this change and I thought it was a good idea.  This enhancement was briefly brought up in #7023 and confirmed it should be added. Was planning on submitting this change with a change to make the entire paginate component sticky, thus eliminating the need to have it replicated at the bottom of the page, but after talking to Remington about it we've decided that it's better to wait until his changes for the sidebar go in first since that will refactor the layout of the page significantly. 

See demo here: 

https://user-images.githubusercontent.com/50851526/132015117-e9faf350-dbe2-4b10-8baf-3bf3e84d88d9.mp4

@rbreeze

### CODE REVIEW

**Review by rbreeze — CHANGES_REQUESTED** [review]
Looks great, thanks! Just a couple comments: 

1. I think it might be a good idea to store this preference in local storage, just like the filter show/hide preference.

2. I think we could choose a better icon that indicates what is to be shown/hidden, rather than just an arrow. We could change the icon's color depending on if it is shown/hidden. A few options I found: 

- https://fontawesome.com/v5.15/icons/map?style=solid
- https://fontawesome.com/v5.15/icons/minus-square?style=solid
- https://fontawesome.com/v5.15/icons/ruler-horizontal?style=solid
- https://fontawesome.com/v5.15/icons/window-minimize?style=solid

Personally I like the `ruler` icon best

**Comment by reginapizza** [comment]
@rbreeze I'm not sure if I'm understand where you're suggestion the hide/show option would go, could you point it out so I'm sure? I don't mind changing the icon to something else, I like the ruler. Where you thinking it would it always be just the ruler regardless of it being in a shown/hidden state?

**Comment by rbreeze** [comment]
Screen Shot 2021-09-08 at 1 46 57 PM

Forgive the crude mockup, but I was thinking the ruler button could go where the blue outline is (to the right of the search bar) and be `argo-button--o` when hidden and normal `argo-button` when shown. The green outline is where I thought the status bar could go

**Comment by rbreeze** [comment]
Screen Shot 2021-09-08 at 1 48 53 PM
Screen Shot 2021-09-08 at 1 48 50 PM


An example of the button behavior I'm thinking of is the `Unschedulable` button in the Pod View

**Comment by rbreeze** [comment]
Actually, now that I'm thinking about it, it may look better to put the status bar to the right of the search bar instead of beneath. Feel free to experiment with both

**Comment by reginapizza** [comment]
Current implementation is now set as a preference instead of just a toggle and looks like so: 

https://user-images.githubusercontent.com/50851526/133341794-586c7bb5-15a5-4308-b593-e84704e526bd.mp4

File: ui/src/app/shared/services/view-preferences-service.ts [hunk 1]
```diff
@@ -21,19 +21,20 @@ export interface PodViewPreferences {
...
-        return [pref.clustersFilter, pref.healthFilter, pref.labelsFilter, pref.namespacesFilter, pref.projectsFilter, pref.reposFilter, pref.syncFilter].reduce(
-            (count, filter) => {
-                if (filter && filter.length > 0) {
-                    return count + 1;
-                }
-                return count;
-            },
-            0
-        );
+        return [pref.clustersFilter, pref.healthFilter, pref.labelsFilter, pref.namespacesFilter, pref.projectsFilter].reduce((count, filter) => {
```
Comment (rbreeze) [inline ui/src/app/shared/services/view-preferences-service.ts:32]: it looks like `pref.reposFilter` and `pref.syncFilter` got removed here. Was this intentional?

**Review by rbreeze — CHANGES_REQUESTED** [review]
LGTM, just one small comment

File: ui/src/app/shared/services/view-preferences-service.ts (hunk 1 above)
Comment (reginapizza) [reply ui/src/app/shared/services/view-preferences-service.ts]: yep that was definitely an accidental deletion, nice catch. I've updated that now!

**Review by rbreeze — APPROVED** [review]
LGTM

**Pull request merged**